packages/x-date-pickers-pro/src/DateTimeRangePicker/DateTimeRangePickerToolbar.tsx

```diff
@@ -76,24 +76,46 @@ const DateTimeRangePickerToolbarStart = styled(DateTimePickerToolbar, {
   name: 'MuiDateTimeRangePickerToolbar',
   slot: 'StartToolbar',
   overridesResolver: (_, styles) => styles.startToolbar,
-})<DateTimeRangePickerStartOrEndToolbarProps<any>>(({ ownerState }) => ({
+})<DateTimeRangePickerStartOrEndToolbarProps<any>>({
```

Comment:
```suggestion
})<DateTimeRangePickerStartOrEndToolbarProps<any> & { ownerState: DateTimeRangePickerStartOrEndToolbarProps<any> }>({
```

This would make sure that the styled-component receives `ownerState`.

Reply:
`DateTimeRangePickerStartOrEndToolbarProps` already contains it :+1:

Reply:
OK the problem was that `ownerState` was optional in `DateTimeRangePickerStartOrEndToolbarProps`

Reply:
And I'll remove the explicit typing from all the callback usage of `props`

Reply:
But the right API here is :

```ts
      props: ({ toolbarVariant }: DateTimeRangePickerStartOrEndToolbarProps<any>) => toolbarVariant !== 'desktop',
```

In which case I do need the typing because the types equals `props` and not `props & ownerState` (it should probably be fixed)